include/circt/Dialect/HW/HWAttributes.td

```diff
@@ -313,4 +313,39 @@ def EnumFieldAttr : AttrDef<HWDialect, "EnumField"> {
   }];
 }
 
+// An attribute to indicate the lint waivers that need to be emitted for the
+// module.
+def WaiverAttr : AttrDef<HWDialect, "Waiver"> {
```

Comment:
Waiver seems to be very specific to SV, so maybe SV is better to place to put this?